### PR TITLE
feat(utxo-lib): add support for p2tr in bitgo.signInput2Of3()

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Issue: BG-35578

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/signature.ts [hunk 1]
```diff
@@ -55,14 +56,21 @@ export interface ParsedSignatureScript2Of3 extends ParsedSignatureScript {
   pubScript: Buffer;
 }
 
-export function getDefaultSigHash(network: Network): number {
+export interface ParsedSignatureScriptTaproot extends ParsedSignatureScript {
+  // P2TR tapscript spends are for 2-of-2 multisig scripts
+  signatures: [Buffer | 0, Buffer | 0]; // missing signature is encoded as OP_0
```
Comment (brandonblack) [inline modules/utxo-lib/src/bitgo/signature.ts:61]: @sangaman ran into a situation where missing signature was encoded as `Buffer.of()` instead of `0`.

Comment (brandonblack) [reply modules/utxo-lib/src/bitgo/signature.ts:61]: Don't think the difference matters here, come to think of it.

Comment (OttoAllmendinger) [reply modules/utxo-lib/src/bitgo/signature.ts:61]: I think `Buffer.of()` is the correct encoding

**Review by brandonblack — DISMISSED** [review]

File: modules/utxo-lib/src/bitgo/signature.ts [hunk 2]
```diff
@@ -316,29 +378,80 @@ export function getSignatureVerifications(
...
   const publicKeys = parsedScript.publicKeys.filter(
-    (buf) => verificationSettings.publicKey === undefined || verificationSettings.publicKey.equals(buf)
+    (buf) =>
+      verificationSettings.publicKey === undefined ||
+      verificationSettings.publicKey.equals(buf) ||
+      verificationSettings.publicKey.slice(1).equals(buf)
   );
 
+  if (verificationSettings.publicKey) {
+    debugger;
```
Comment (tylerlevine) [inline modules/utxo-lib/src/bitgo/signature.ts:388]: should be removed

**Review by tylerlevine — CHANGES_REQUESTED** [review]
looks really good, just remove the `debugger` statement and I'm good to go

**Review by tylerlevine — DISMISSED** [review]

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**